### PR TITLE
Use patch found in the PATH

### PR DESCRIPTION
**Pull request by sjackman**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/master/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/homebrew/pull/49031).
- [x] Have you successfully run `brew tests` with your changes locally?

### CODE REVIEW

**Comment by MikeMcQuaid** [comment]
I'd rather we avoid any more of these until environment filtering is on by default. I think then we can just remove the hardcoded `/usr/bin` pretty much everywhere. Thoughts?

**Comment by sjackman** [comment]
> I think then we can just remove the hardcoded `/usr/bin` pretty much everywhere.

That would be awesome.

**Comment by sjackman** [comment]
Before I read your comment, I came up with a better solution using `with_system_path`. It may still be worth applying, even in advance of environment filtering. After environment filtering, we can probably get rid of `with_system_path` and its callers.

**Comment by MikeMcQuaid** [comment]
> Before I read your comment, I came up with a better solution using with_system_path

I think this kinda changes the meaning of `with_system_path`, though.

File: Library/Homebrew/patch.rb [hunk 1]
```diff
@@ -64,9 +64,9 @@ def contents; end
 
   def apply
     data = contents.gsub("HOMEBREW_PREFIX", HOMEBREW_PREFIX)
-    cmd = "/usr/bin/patch"
+    cmd = "patch"
```
Comment (MikeMcQuaid) [inline Library/Homebrew/patch.rb]: Let's try just changing this to `patch` below and not use a variable (i.e. remove the `/usr/bin`). If it goes 💥 
I'll revert it.

Comment (sjackman) [reply Library/Homebrew/patch.rb]: I got rid of the `cmd` variable. Did you mean also get rid of `with_system_path`?

**Comment by sjackman** [comment]
>> Before I read your comment, I came up with a better solution using with_system_path

> I think this kinda changes the meaning of with_system_path, though.

You man not find the desired utility in `/usr/bin` or `/bin` on a Linux system, where it makes sense to use the brewed version of the utility, if it's installed. It could be renamed to `with_sanitized_path`.

File: Library/Homebrew/patch.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Library/Homebrew/patch.rb]: @sjackman Yep; make this just remove `/usr/bin/` and the variable usage.

Comment (sjackman) [reply Library/Homebrew/patch.rb]: Done. I also replaced `IO.popen` with `Utils.popen_write`.

**Comment by MikeMcQuaid** [comment]
Thanks again @sjackman!

**Comment by sjackman** [comment]
No worries. Thanks for merging, Mike.

**Comment by MikeMcQuaid** [comment]
@sjackman Now that environment filtering is on by default @sjackman you can feel free to do this anywhere else that makes sense (and remove a bunch of `with_system_path` usage, too.

**Comment by sjackman** [comment]
Fantastic!

**Pull request merged**